File: CS/CustomPaintEvent/Form1.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using System.Drawing.Drawing2D;
using System.Collections.Generic;
using DevExpress.XtraCharts;
// ...

namespace CustomPaintEvent {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void chartControl1_CustomPaint(object sender, CustomPaintEventArgs e) {
            ChartControl chart = (ChartControl)sender;
            Series series = chart.Series[0];

            Point[] screenPoints = new Point[series.Points.Count];

            for (int i = 0; i < series.Points.Count; i++) {
                screenPoints[i] =
                    ((XYDiagram2D)chart.Diagram).DiagramToPoint(series.Points[i].Argument,
                    series.Points[i].Values[0]).Point;
            }

            Graphics g = e.Graphics;
            g.SmoothingMode = SmoothingMode.AntiAlias;
            g.FillPolygon(new SolidBrush(Color.LightGreen), screenPoints);
            g.DrawPolygon(new Pen(Color.DarkGreen, 2), screenPoints);
        }

    }
}

File: VB/CustomPaintEvent/Form1.vb
Imports Microsoft.VisualBasic
Imports System.Drawing
Imports System.Windows.Forms
Imports System.Drawing.Drawing2D
Imports DevExpress.XtraCharts
' ...

Namespace CustomPaintEvent
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub chartControl1_CustomPaint(ByVal sender As Object, ByVal e As CustomPaintEventArgs) Handles chartControl1.CustomPaint
			Dim chart As ChartControl = CType(sender, ChartControl)
			Dim series As Series = chart.Series(0)

			Dim screenPoints(series.Points.Count - 1) As Point

			For i As Integer = 0 To series.Points.Count - 1
				screenPoints(i) = (CType(chart.Diagram, XYDiagram2D)).DiagramToPoint(series.Points(i).Argument, series.Points(i).Values(0)).Point
			Next i

			Dim g As Graphics = e.Graphics
			g.SmoothingMode = SmoothingMode.AntiAlias
			g.FillPolygon(New SolidBrush(Color.LightGreen), screenPoints)
			g.DrawPolygon(New Pen(Color.DarkGreen, 2), screenPoints)
		End Sub

	End Class
End Namespace
